packages/block-library/src/navigation-link/edit.js

```diff
@@ -598,7 +579,7 @@ export default function NavigationLinkEdit( {
 							onClose={ () => setIsLinkOpen( false ) }
 							anchor={ popoverAnchor }
 							hasCreateSuggestion={ userCanCreate }
-							onRemove={ removeLink }
+							onRemove={ () => removeBlocks( clientId ) }
```

Comment:
I don't think we should do this in _this_ PR. It has knock on consequences. For example, it will not allow folks to add empty link items which I believe was functionality that was specifically requsted by users in the past.

The use case is that you might want to build out the structure of a Nav without necessarily adding links to each item. 

With this change any nav link block that doesn't add a link using the initial Link UI popup will be removed.

Happy to consider this but I think it should be in a separate PR to avoid over burdening this PR 🙏